src/Components/Components/src/NavigationRedirectHelper.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Helper class to throw a <see cref="NavigationException"/>.
+/// </summary>
+public static class NavigationRedirectHelper
+{
+    private const string RedirectExceptionMessage =
+        "A navigation was initiated during static rendering. " +
+        "This exception is not an error, but instead signals to the framework that a redirect should occur. " +
+        "It should not be caught by application code.";
+
+    /// <summary>
+    /// Creates an absolute URI and throws a new <see cref="NavigationException"/>.
+    /// </summary>
+
+    [DoesNotReturn]
+    public static void ThrowNavigationExceptionForRedirect(NavigationManager navigationManager, string uri)
+    {
+        var absoluteUriString = navigationManager.ToAbsoluteUri(uri).AbsoluteUri;
+        throw new NavigationException(absoluteUriString, RedirectExceptionMessage);
+    }
+}
```

Comment:
We don't need a helper class to replace what otherwise is two lines of code that get called in two locations. The cost of adding new APIs is not worth the "savings" that we might get. Specially since this exception is entirely thrown by the framework and no one else.

Reply:
I have removed the helper class